src/main/java/org/folio/circulation/domain/Request.java

```diff
@@ -132,12 +133,16 @@ boolean isFor(User user) {
 
   @Override
   public String getItemId() {
-    return requestRepresentation.getString("itemId");
+    return requestRepresentation.getString(ITEM_ID);
   }
 
   public Request withItem(Item newItem) {
-    return new Request(requestRepresentation,cancellationReasonRepresentation, newItem, requester, proxy,
-      loan == null ? null : loan.withItem(newItem),pickupServicePoint);
+    // NOTE: this is null in RequestsAPIUpdatingTests.replacingAnExistingRequestRemovesItemInformationWhenItemDoesNotExist test 
```

Comment:
It was unexcpected to me. The item may be new but the JSON itemRepresentation did not have an id. I am not sure if this was intended. @marcjohnson-kint can you provide some insight?

Reply:
@mreno-EBSCO  What is expected functionality?

Reply:
I suspect this is the cumulative effect of convenient extensions to the classes to carry around related objects :-(

Reply:
The item for the request and the item should be the same. It appears this code tries to ensure that. And be defensive so if the loan isn't provided, don't blow up, because the loan isn't needed in all situations.

I suspect that in reality there really isn't this kind of relationship between a request and loan, and instead the loan should be moved out of the request class (I vaguely recall comments to this effect on a previous pull request).

Reply:
I'm a little surprised the `loan` isn't null in more circumstances

Reply:
What I meant, is that was it expected the itemId could be null. I was concerned we may have a bad test case where we are setting it to null as a side effect, but it seemed pretty clear that the test was doing this on purpose. I just thought it might be better to cite the workflow that causes this to be null as opposed to a test case.

Reply:
I have noted the test that fails. So we can track down the cause of the item not having its id in the JSON representation. This may be byond the scope of this card.

Reply:
Got it. It is not expected that itemId can be null here and likely a bad test case or a bug in the code.

Reply:
I think I'm lost, but that might be ok. Where is there an item ID referenced in that code?

Reply:
Sorry Marc, probably my fault as I put the comment on the "note" line. I assumed the note was referring to the line below it where this code is:
```
    // NOTE: this is null in RequestsAPIUpdatingTests.replacingAnExistingRequestRemovesItemInformationWhenItemDoesNotExist test
    if (newItem.getItemId() != null) {
      requestRepresentation.put(ITEM_ID, newItem.getItemId());
    }
```

I was just concerned that the `if` was only in response to a failed test (which could be broken) and whether or not we needed this code here under normal operation. I just wanted some clarification about the note.

Reply:
That explains why my comments make no sense. I was referring to the line above :-(

Reply:
Ok, switching my comments. That is weird. 

That seems to be a hangover from the use of `JSONObject` within the domain. We likely shouldn't be overwriting the `itemId` in the JSON. And any mismatch should likely cause an error.

This is another indicator that the convenient design of passing the JSON representation through the domain has likely overstayed it's welcome and needs removing.

That is way beyond this issue. I raised [CIRC-369](https://issues.folio.org/browse/CIRC-369) to start addressing this (when we get some time to improve the internal design, see [umbrella](https://issues.folio.org/browse/CIRC-346)